src/main/java/de/mediathekview/mserver/crawler/mdr/tasks/MdrFilmTask.java

```diff
@@ -0,0 +1,122 @@
+import java.net.URL;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+import mServer.crawler.CrawlerTool;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jsoup.nodes.Document;
+import org.xml.sax.SAXException;
+
+public class MdrFilmTask extends AbstractDocumentTask<Film, CrawlerUrlDTO> {
+
+  private static final Logger LOG = LogManager.getLogger(MdrFilmTask.class);
+
+  private final String baseUrl;
+  private final MdrFilmPageDeserializer filmPageDeserializer;
+
+  public MdrFilmTask(AbstractCrawler aCrawler,
+      ConcurrentLinkedQueue<CrawlerUrlDTO> aUrlToCrawlDtos,
+      final String aBaseUrl) {
+    super(aCrawler, aUrlToCrawlDtos);
+
+    baseUrl = aBaseUrl;
+    filmPageDeserializer = new MdrFilmPageDeserializer(baseUrl);
+  }
+
+  @Override
+  protected void processDocument(CrawlerUrlDTO aUrlDto, Document aDocument) {
+
+    Set<CrawlerUrlDTO> filmEntries = filmPageDeserializer.deserialize(aDocument);
+
+    for (CrawlerUrlDTO filmEntry : filmEntries) {
+      try {
+        Optional<MdrFilmXmlHandler> mdrFilmXmlHandler = loadFilmXml(filmEntry);
+        if (mdrFilmXmlHandler.isPresent()) {
+          Film film = createFilm(mdrFilmXmlHandler.get(), filmEntry);
+          taskResults.add(film);
+          crawler.incrementAndGetActualCount();
+          crawler.updateProgress();
+        } else {
+          LOG.error("MdrFilmTask: no film found for url " + aUrlDto.getUrl());
+          crawler.incrementAndGetErrorCount();
+          crawler.updateProgress();
+        }
+      } catch (MalformedURLException e) {
+        LOG.error("MdrFilmTask: error reading video infos " + aUrlDto.getUrl(), e);
+        crawler.incrementAndGetErrorCount();
+        crawler.updateProgress();
+      }
+    }
+  }
+
+  @Override
+  protected AbstractRecrusivConverterTask<Film, CrawlerUrlDTO> createNewOwnInstance(
+      ConcurrentLinkedQueue<CrawlerUrlDTO> aElementsToProcess) {
+    return new MdrFilmTask(crawler, aElementsToProcess, baseUrl);
+  }
+
+  private Optional<MdrFilmXmlHandler> loadFilmXml(final CrawlerUrlDTO aXmlUrl) {
+    try {
+
+      SAXParserFactory factory = SAXParserFactory.newInstance();
+      SAXParser saxParser = factory.newSAXParser();
+      MdrFilmXmlHandler handler = new MdrFilmXmlHandler();
+      saxParser.parse(aXmlUrl.getUrl(), handler);
+
+      return Optional.of(handler);
+
+    } catch (SAXException | IOException | ParserConfigurationException e) {
+      LOG.error(String.format("Error loading xml document \"%s\".", aXmlUrl.getUrl()), e);
+    }
+
+    return Optional.empty();
+  }
+
+  private Film createFilm(final MdrFilmXmlHandler aFilmXmlHandler, final CrawlerUrlDTO aWebsiteUrl) throws MalformedURLException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "aWebsiteUrl". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)